crates/rolldown/src/ast_scanner/side_effect_detector/utils.rs

```diff
@@ -22,49 +22,27 @@ impl<'a> SideEffectDetector<'a> {
   ///
   /// let c = 1;
   ///
-  /// let d = 1; /* invalid comment for `e` */
+  /// let d = 1; /* valid comment for `e` */
   /// let e = 2
   /// ```
   /// Derived from https://github.com/oxc-project/oxc/blob/147864cfeb112df526bb83d5b8671b465c005066/crates/oxc_linter/src/utils/tree_shaking.rs#L204
   pub fn leading_comment_for(&self, span: Span) -> Option<(&Comment, &str)> {
     let comment = self.trivias.comments_range(..span.start).next_back()?;
 
     let comment_text = comment.span.source_text(self.source);
-
-    // If there are non-whitespace characters between the `comment`` and the `span`,
+    // If there are non-whitespace characters between the `comment` and the `span`,
     // we treat the `comment` not belongs to the `span`.
-    let only_whitespace = Span::new(comment.span.end, span.start)
-      .source_text(self.source)
-      .strip_prefix("*/") // for multi-line comment
-      .is_some_and(|s| s.trim().is_empty());
-
-    if !only_whitespace {
-      return None;
-    }
-
-    // Next step, we need make sure it's not the trailing comment of the previous line.
-    let mut current_line_start = span.start as usize;
-    for c in self.source[..span.start as usize].chars().rev() {
-      if c == '\n' {
-        break;
+    let range_text = Span::new(comment.span.end, span.start).source_text(self.source);
+    let only_whitespace = match comment.kind {
```

Comment:
Should consider the `SingleLine` scenario, or it just returns `None` because `strip_prefix` would always fail.